### PR TITLE
Extrapolate landice variable

### PR DESCRIPTION
**Pull request by trhille**
Add a script to extrapolate landice variables into regions of missing or invalid data using nearest-neighbor or inverse-distance weighted methods.

### CODE REVIEW

**Comment by xylar** [comment]
@trhille, CI is breaking for reasons unrelated to your script.  Someone accidentally released `matplotlib` 3.9.0 on conda-forge a few days back and did a bunch of upgrading to support it.  It turned out they jumped the gun and they moved that release off the main conda-forge channel so now it can't be installed without a lot of extra hassle.  I was hoping the actual release would happen but that also doesn't seem to have happened.

Anyway, I'll fix it if it becomes critical before they actually release the new version.

**Comment by trhille** [comment]
Thanks, @xylar. This script has been lying around on a branch for years, and we decided that it has ended up with enough use cases that we'll merge it, but it's not at all time-sensitive.

**Comment by xylar** [comment]
I'm going to close and re-open to cause CI to restart.

**Review by matthewhoffman — CHANGES_REQUESTED** [review]
@trhille , thanks for formalizing this script.  I have a few comments - feel free to take or leave.  The script works as-is and I don't want to drag this out.

File: landice/mesh_tools_li/extrapolate_variable.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        for iCell in range(nCells):
+            for n in range(nEdgesOnCell[iCell]):
+                jCell = cellsOnCell[iCell, n] - 1
+                if (groundedMask[jCell] == 1):
+                    keepCellMask[iCell] = 1
+                    continue
+        # ensure zero muFriction does not get extrapolated
+        keepCellMask *= (varValue > 0)
+        # Get rid of invalid values
+        keepCellMask *= (varValue < 1.e12)
```
[the remark this answers is not in the document]
Comment (trhille) [reply landice/mesh_tools_li/extrapolate_variable.py:92]: I think it's safe, but will add it as a constant at the top of the script as with the densities.

File: landice/mesh_tools_li/extrapolate_variable.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    if has_vertical_dim == True:
+        print(f"Processing vertical level number {v}")
+        varValue = dataset.variables[var_name][0, :, v]
+    else:
+        varValue = dataset.variables[var_name][0, :]
+
+    # Define region of good data to extrapolate from.
+    # Different methods for different variables
+    if var_name in ["effectivePressure", "beta", "muFriction"]:
+        groundedMask = (thickness > (-1028.0 / 910.0 * bed))
```
[the remark this answers is not in the document]
Comment (trhille) [reply landice/mesh_tools_li/extrapolate_variable.py]: Good catch. I have now changed this to raise an exception with a helpful message if thickness and/or bedTopography are missing. Addressed in https://github.com/MPAS-Dev/MPAS-Tools/pull/576/commits/8f7ef92d3e20cdebe603461440cd1977ce921e24.

File: landice/mesh_tools_li/extrapolate_variable.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (trhille) [reply landice/mesh_tools_li/extrapolate_variable.py]: Addressed in https://github.com/MPAS-Dev/MPAS-Tools/pull/576/commits/8f7ef92d3e20cdebe603461440cd1977ce921e24

**Review by matthewhoffman — APPROVED** [review]

**Pull request merged**